### PR TITLE
Added basic tests for coordinates

### LINKED ISSUE #26: Unit Tests for Coordinates

### PR DESCRIPTION
**Pull request by klanmiko**
Closes #26 
Tests coordinate equality, Direction to coordinate addition, and coordinate to coordinate condition. Also tests implicit conversion from CellCoordinate to NodeCoordinate when the two different types are added together.

### CODE REVIEW

File: Test/Test-Coordinate.cpp [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        }
+    }
+}
+SCENARIO("Equivalence of coordinates") {
+    GIVEN("A NodeCoordinate") {
+        NodeCoordinate a = NodeCoordinate(2, 5);
+        WHEN("It is compared with itself") {
+            THEN("It is equal") {
+                REQUIRE(a == a);
+                REQUIRE(!(a != a));
```
Comment (blhough) [inline Test/Test-Coordinate.cpp]: very minor thing but there is a `REQUIRE_FALSE` macro to make it more succinct

**Comment by blhough** [comment]
Very nice! Would like to see operator*, default constructors, isCell and isPost in there as well. Those can be part of another Issue and PR if you would like though.

**Comment by klanmiko** [comment]
Added the remaining tests as specified, also started using require_false

**Comment by blhough** [comment]
Beautiful! Any idea what you want to work on next?

**Pull request merged**